### PR TITLE
Gain Experience in "Automatic passenger deletion" logic (#806)

### PR DESCRIPTION
**Pull request by Envoidener**
Also added AffectedByVeterancy boolean

### CODE REVIEW

File: src/Ext/Techno/Body.cpp [hunk 1]
```diff
@@ -171,8 +171,25 @@ void TechnoExt::ExtData::EatPassengers()
 			{
 				// Setting & start countdown. Bigger units needs more time
 				int passengerSize = pTypeExt->PassengerDeletion_Rate;
-				if (pTypeExt->PassengerDeletion_Rate_SizeMultiply && pPassenger->GetTechnoType()->Size > 1.0)
+				if (pTypeExt->PassengerDeletion_Rate_SizeMultiply &&
+					pPassenger->GetTechnoType()->Size > 1.0)
```
Comment (Metadorius) [inline src/Ext/Techno/Body.cpp:175]:
```suggestion
				if (pTypeExt->PassengerDeletion_Rate_SizeMultiply
					&& pPassenger->GetTechnoType()->Size > 1.0)
```

**Review by chaserli — CHANGES_REQUESTED** [review]
Do it in #654 instead

**Comment by Metadorius** [comment]
Yeah it might be good to rebase it onto #654 for consistency.

**Comment by ek-development** [comment]
Might as well close this PR as most of this is redundant; will continue on #654

**Comment by Envoidener** [comment]
Might as well close this PR as most of this is redundant; will continue on #654

**Pull request closed without merging**